capz/run-capz-e2e.sh

```diff
@@ -151,6 +161,29 @@ apply_workload_configuraiton(){
     kubectl apply -f "${CAPZ_DIR}"/templates/addons/metrics-server/metrics-server.yaml
 }
 
+apply_cloud_provider_azure() {
+    echo "KUBERNETES_VERSION = ${KUBERNETES_VERSION}"
+
+    echo "Building cloud provider images"
+    # shellcheck disable=SC1091
+    "${CAPZ_DIR}/hack/ensure-acr-login.sh"
+    # shellcheck disable=SC1091
+    source "${CAPZ_DIR}/scripts/ci-build-azure-ccm.sh" || false
+    trap run_capz_e2e_cleanup EXIT # reset the EXIT trap since ci-build-kubernetes.sh also sets it.
```

Comment:
I think this means the trap from ci-build-azure-ccm.sh won't run. If we don't source the script does that work?  

If not should we make sure we do what ever clean up?  Looking at the source it seems like its just doing some general local cleanup and this pod should get removed so shouldn't be a big deal, _unless_ something changes in the future.

Reply:
we need need to source the script. 
Let me look at this again and see if the the trap from ci-build-azure-ccm.sh will run or not

Reply:
The trap does not get called :(
To work around this I have opened https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3686 and I will call the cleanup functions for each script inside the cleanup method in this script

Reply:
fixed the issue by calling the cleanup methods from the scripts we need to source explicitly